lib/veri_traci.ml

```diff
@@ -0,0 +1,79 @@
+open Bap.Std
+open Bap_traces.Std
+open Monad.State
+
+type 'a u = 'a Bil.Result.u
+type event = Trace.event
+
+let stub = fun _ -> return () 
+
+let mem_of_arch arch = 
+  let (module T : Target) = target_of_arch arch in
+  T.CPU.mem
+
+class context trace =
+  object(self:'s)
+    inherit Bili.context
+  end
+
+let mem_operands mv = 
+  let width = Word.bitwidth (Move.data mv) in
+  Size.of_int_opt width |> function
+  | None -> 
+    let er = Printf.sprintf "wrong size for memory operand: %d" width in
+    failwith er
+  | Some sz ->
+    let addr = Bil.int (Move.cell mv) in
+    let data = Bil.int (Move.data mv) in
+    addr, data, sz
+
+class ['a] t arch = 
+  let mem = mem_of_arch arch in 
+  let mem' = Bil.var mem in
+  object(self)
+    constraint 'a = #context
+    inherit ['a] Bili.t
+
+    method eval_memory_store mv = 
+      let addr, data, size = mem_operands mv in
+      get () >>= fun ctxt ->
+      let endian = Arch.endian arch in
+      self#eval_store ~mem:mem' ~addr data endian size >>= fun r ->
+      get () >>= fun ctxt -> self#update mem r
+
+    method eval_register_write mv = 
+      self#eval_move (Move.cell mv) (Bil.int (Move.data mv))
+
+    method eval_register_read: var move -> 'a u = stub
+    method eval_memory_load: addr move -> 'a u = stub
```

Comment:
`eval_memory_load` shouldn't be empty. It should dispatch to `eval_load` (that may evaluate to `bot`, but nobody cares as the result is not used).